packages/core/src/domain/context/storedContextManager.ts

```diff
@@ -0,0 +1,87 @@
+import { computeBytesCount } from '../../tools/utils/byteUtils'
+import { addEventListener, DOM_EVENT } from '../../browser/addEventListener'
+import type { Context } from '../../tools/serialisation/context'
+import type { Configuration } from '../configuration'
+import type { ContextManager } from './contextManager'
+import { createContextManager } from './contextManager'
+import type { CustomerDataType } from './contextConstants'
+
+const CONTEXT_STORE_KEY_PREFIX = '_dd_c'
+
+export function createStoredContextManager(
+  configuration: Configuration,
+  productKey: string,
+  customerDataType: CustomerDataType,
+  computeBytesCountImpl = computeBytesCount
+): ContextManager {
+  const storageKey = buildStorageKey(productKey, customerDataType)
+  const contextManager = createContextManager(customerDataType, computeBytesCountImpl)
+
+  synchronizeWithStorage()
+  addEventListener(configuration, window, DOM_EVENT.STORAGE, ({ key }) => {
+    if (storageKey === key) {
+      synchronizeWithStorage()
+    }
+  })
+
+  return {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Instead of implementing every methods, it could be simpler to provide a "onChange" callback to the base `createContextManager` (or return an observer)